NPE when method with Mono/Flux can return null

Seems it's not checked when introspecting methods that sometimes it can be null.

For example, its failed for me with own reactor library reactor-rabbitmq:
```
Caused by: java.lang.NullPointerException
	at reactor.rabbitmq.ReceiverOptions.getConnectionMono(ReceiverOptions.java:80)
	at reactor.rabbitmq.Receiver.<init>(Receiver.java:65)
	at reactor.rabbitmq.RabbitFlux.createReceiver(RabbitFlux.java:25)
```

```
public Receiver(ReceiverOptions options) {
        this.privateConnectionSubscriptionScheduler = options.getConnectionSubscriptionScheduler() == null;
        this.connectionSubscriptionScheduler = options.getConnectionSubscriptionScheduler() == null ?
            createScheduler("rabbitmq-receiver-connection-subscription") : options.getConnectionSubscriptionScheduler();
        this.connectionMono = options.getConnectionMono() != null ? options.getConnectionMono() :
            Mono.fromCallable(() -> options.getConnectionFactory().newConnection())
                .doOnSubscribe(c -> hasConnection.set(true))
                .subscribeOn(this.connectionSubscriptionScheduler)
                .cache();
    }
```

where options.getConnectionMono() can be null its failed.

Any suggestion workaround for this situation?

--- a/reactor-tools/src/main/java/reactor/tools/agent/ReactorDebugAgent.java
+++ b/reactor-tools/src/main/java/reactor/tools/agent/ReactorDebugAgent.java
@@ -149,7 +149,13 @@ public class ReactorDebugAgent {
 					}
 				};
 
-				cr.accept(classVisitor, 0);
+				try {
+					cr.accept(classVisitor, 0);
+				}
+				catch (Throwable e) {
+					e.printStackTrace();
+					throw e;
+				}
 
 				if (!changed.get()) {
 					return null;

--- a/reactor-tools/src/main/java/reactor/tools/agent/ReturnHandlingMethodVisitor.java
+++ b/reactor-tools/src/main/java/reactor/tools/agent/ReturnHandlingMethodVisitor.java
@@ -80,7 +80,7 @@ class ReturnHandlingMethodVisitor extends MethodVisitor {
 
     @Override
     public void visitMethodInsn(int opcode, String owner, String name, String descriptor, boolean isInterface) {
-        super.visitMethodInsn(opcode, owner,name, descriptor, isInterface);
+        super.visitMethodInsn(opcode, owner, name, descriptor, isInterface);
 
         if (!checkpointed && CallSiteInfoAddingMethodVisitor.isCorePublisher(owner)) {
             String returnType = Type.getReturnType(descriptor).getInternalName();
@@ -94,6 +94,11 @@ class ReturnHandlingMethodVisitor extends MethodVisitor {
     public void visitInsn(int opcode) {
         if (!checkpointed && Opcodes.ARETURN == opcode) {
             changed.set(true);
+
+            super.visitInsn(Opcodes.DUP);
+            Label atReturn = new Label();
+            super.visitJumpInsn(Opcodes.IFNULL, atReturn); // skip if null
+
             String callSite = String.format(
                     "%s.%s(%s:%d)",
                     currentClassName.replace("/", "."), currentMethod, currentSource, currentLine
@@ -106,6 +111,9 @@ class ReturnHandlingMethodVisitor extends MethodVisitor {
                     "(Ljava/lang/String;)L" + returnType + ";",
                     false
             );
+
+            super.visitLabel(atReturn);
+            super.visitFrame(Opcodes.F_SAME1, 0, null, 1, new Object[] { returnType });
         }
 
         super.visitInsn(opcode);

--- a/reactor-tools/src/test/java/reactor/tools/agent/ReactorDebugAgentTest.java
+++ b/reactor-tools/src/test/java/reactor/tools/agent/ReactorDebugAgentTest.java
@@ -77,9 +77,21 @@ public class ReactorDebugAgentTest {
 				.startsWith("GroupedFlux.map ⇢ reactor.tools.agent.ReactorDebugAgentTest.shouldWorkWithGroupedFlux(ReactorDebugAgentTest.java:" + (baseline + 1));
 	}
 
+	// See https://github.com/reactor/reactor-tools/issues/7
+	@Test
+	public void shouldHandleNullReturnValue() {
+		Mono<Integer> mono = methodReturningNullMono();
+
+		assertThat(mono).isNull();
+	}
+
 	static final int methodReturningMonoBaseline = getBaseline();
 	private Mono<Integer> methodReturningMono(Mono<Integer> mono) {
 		return mono;
+	}
+
+	private Mono<Integer> methodReturningNullMono() {
+		return null;
 	}
 
 	private static int getBaseline() {

